### DISCUSSION TITLE
Prompt to close side panel is clicking outside the panel

### DISCUSSION THREAD

**Discussion by benaunt** [Budibase/budibase#13401] [Help] [answered]
Hi,

We are finding users accidently close out of forms in side panels easily by accidentality clicking outside the panel, especially if there are drop downs in the form.

Is there anyway to require the user to click a save/close button within the panel to close it or at least trigger a prompt to confirm to close then panel when clicking outside of it?

Our current work around is to force the panel width to be larger using an embed component, then set padding inside it to provide more room to de-select drop downs.

Thanks

**Comment by mikesealey** [comment, marked as answer]
Hi @benaunt 

There's a solution for this issue currently being worked on. 

The first addition will be a toggle to decide if clicking outside of the side-panel will close it or not. The second piece to the puzzle is the option to add actions that run when the side panel closes.  

These two additional settings should help you manage any accidental stray clicks from spoiling a user's experience.

[Link to PR](https://github.com/Budibase/budibase/pull/13488)

**Comment by mikesealey** [comment]
Hi @benaunt 

If you update to the latest version of Budibase, you should now see some settings available in the sidepanel menu. Ignore Clicks Outside will do just that - clicking outside of the side panel will no longer close it (with the exception of navigation links). 

You can also configure actions to run when the side panel closes.

![image](https://github.com/Budibase/budibase/assets/39269380/922aea26-12dc-4665-aaff-7ef5e095074f)
